src/Domain.Linnapps/Products/ProductRange.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Linn.Products.Domain.Linnapps.Products
+{
+    using System;
+    using System.ComponentModel.DataAnnotations.Schema;
+
+    using Linn.Common.Domain.Exceptions;
+
+    public class ProductRange
+    {
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; set; }
+
+        public string RangeName { get; set; }
+
+        public string RangeDescription { get; set; }
+
+        public DateTime? DateInvalid { get; set; }
+
+        public void ValidateProductRange()
```

Comment:
Are you doing this for all non nullable db fields? I've not done that but is it worth doing in all the domain classes?

Reply:
I haven't done it everywhere but having done it here it seems a good thing.